### PR TITLE
redirect to latest course standards based on family name

### PR DESCRIPTION
**Pull request by davidsbailey**
Finishes https://codedotorg.atlassian.net/browse/PLAT-1125. To see how this helps us, take a peek at the following PR:  https://github.com/code-dot-org/code-dot-org/pull/41571

## Testing story

new unit test

### CODE REVIEW

File: dashboard/app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -155,6 +155,14 @@ def code
 
   def standards
     unit_group = UnitGroup.get_from_cache(params[:course_name])
+    if !unit_group.present? && UnitGroup.family_names.include?(params[:course_name])
+      redirect_to_course = UnitGroup.all_courses.
+        select {|c| c.family_name == params[:course_name] && c.stable?}.
+        sort_by(&:version_year).
+        last
+      redirect_to "/courses/#{redirect_to_course.name}/standards"
```
Comment (Hamms) [inline dashboard/app/controllers/courses_controller.rb]: can we use a Rails url helper to construct this path, rather than hardcoding it?

**Review by bethanyaconnor — APPROVED** [review]
Code looks good. The jira task you linked also mentioned redirecting for standalone units. Does that already work correctly?

File: dashboard/app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -155,6 +155,14 @@ def code
 
   def standards
     unit_group = UnitGroup.get_from_cache(params[:course_name])
+    if !unit_group.present? && UnitGroup.family_names.include?(params[:course_name])
+      redirect_to_course = UnitGroup.all_courses.
+        select {|c| c.family_name == params[:course_name] && c.stable?}.
+        sort_by(&:version_year).
+        last
```
Comment (Hamms) [inline dashboard/app/controllers/courses_controller.rb]: any reason not to pull this and https://github.com/code-dot-org/code-dot-org/blob/d86cd51e40578bf02853cd377503e7c3507f49bb/dashboard/app/controllers/courses_controller.rb#L73-L76 out into a helper method? Something like `redirect_to_course = UnitGroup.latest_stable(params[:course_name])`

**Review by Hamms — APPROVED** [review]
couple minor nits, but otherwise LGTM!

**Comment by davidsbailey** [comment]
> Code looks good. The jira task you linked also mentioned redirecting for standalone units. Does that already work correctly?

Good question. that does not work already, however I don't see any instances where we link to individual CSF course standards from our marketing pages, so I was planning to skip this, but I could add it if we can think of a reason to. apparently we are still linking to curriculum builder CSF 2020 for CSF standards because that is the only place we have a standards landing page for all of CSF.

**Comment by davidsbailey** [comment]
Great suggestions @Hamms , this cleaned up nicely!

**Pull request merged**